File: userwallet/src/main/java/org/demo/userwallet/service/AuthService.java
package org.demo.userwallet.service;

import org.demo.userwallet.model.User;
import org.demo.userwallet.model.ResponseEntity;
import org.demo.userwallet.model.dto.auth.LoginRequest;

public interface AuthService {
     ResponseEntity register(User user);
     ResponseEntity login(LoginRequest loginRequest);
     void logout(String token);
}

File: frontEnd/src/app/auth/services/auth.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { CookieService } from 'src/app/services/cookie.service';
import { environment } from 'src/environments/environment';

@Injectable({
  providedIn: 'root',
})
export class AuthService {
  constructor(
    private httpClient: HttpClient,
    private cookieService: CookieService
  ) {}

  register(body: any): Observable<any> {
    return this.httpClient.post(`${environment.url}/auth/register`, body);
  }

  login(body: any): Observable<any> {
    return this.httpClient.post(`${environment.url}/auth/login`, body);
  }

  logout(): Observable<any> {
    return this.httpClient.get(`${environment.url}/auth/logout`, {
      headers: {
        Authorization: `Bearer ${this.cookieService.get('token')}`,
      },
    });
  }
}
